modules/cas_cache/ocf_env.c

```diff
@@ -54,6 +55,12 @@ void *env_allocator_new(env_allocator *allocator)
 		BUG_ON(item->used);
 	} else {
 		item = kmem_cache_zalloc(allocator->kmem_cache, GFP_NOIO);
+		if (item) {
+			/* Kmemleak is supposed to track the memory CAS is actually using,
+			   not the container */
+			kmemleak_free(item);
+			kmemleak_alloc(&item->data, allocator->item_size, 1, GFP_NOIO);
```

Comment:
Is this actually necessary? According to kmemleak documentation this will not be detected as a leak as long as &item->data is not lost.

"An allocated block of memory is considered orphan if no pointer to its start address or to any location inside the block can be found by scanning the memory"

Reply:
Won't we get leak report if the scan occurs just between line 62 and line 70? There is no location in memory that holds pointer to `&item->data` in this window.

Reply:
@robertbaldyga In general I think it is possible that some pointers are temporarily stored in registers (or being moved between memory/registers etc) and kmemleak scanning at the wrong moment would identify these allocations as leaked. Making sure that allocation is referenced in memory at all times might not be worth the effort. Instead, since scan is ran on demand, it would be a good practice to run it when pointers are not being passed around.